File: LarkCardKit/Builders/ChartBuilder.cs
using LarkCardKit.Models.Elements;

namespace LarkCardKit.Builders;

/// <summary>
/// 图表构建器
/// </summary>
public class ChartBuilder
{
    private readonly Chart _chart = new();

    /// <summary>
    /// 设置图表类型
    /// </summary>
    public ChartBuilder ChartType(string type)
    {
        _chart.ChartType = type;
        return this;
    }

    /// <summary>
    /// 设置图表数据
    /// </summary>
    public ChartBuilder Data(object data)
    {
        _chart.Data = data;
        return this;
    }

    /// <summary>
    /// 设置图表宽度
    /// </summary>
    public ChartBuilder Width(string width)
    {
        _chart.Width = width;
        return this;
    }

    /// <summary>
    /// 设置元素 ID
    /// </summary>
    public ChartBuilder ElementId(string id)
    {
        _chart.ElementId = id;
        return this;
    }

    /// <summary>
    /// 设置外边距
    /// </summary>
    public ChartBuilder Margin(string margin)
    {
        _chart.Margin = margin;
        return this;
    }

    /// <summary>
    /// 构建 Chart 对象
    /// </summary>
    public Chart Build()
    {
        return _chart;
    }
}

File: TypeScript/LarkCardKit/src/builders/ChartBuilder.ts
import type { Chart } from '../models/elements/Chart';

export class ChartBuilder {
  private chart: Chart = {} as Chart;

  constructor() {
    this.chart.tag = 'chart';
  }

  /**
   * 设置图表类型
   */
  chartType(type: string): ChartBuilder {
    this.chart.chart_type = type;
    return this;
  }

  /**
   * 设置图表数据
   */
  data(data: any): ChartBuilder {
    this.chart.data = data;
    return this;
  }

  /**
   * 设置图表宽度
   */
  width(width: string): ChartBuilder {
    this.chart.width = width;
    return this;
  }

  /**
   * 设置元素 ID
   */
  elementId(id: string): ChartBuilder {
    this.chart.element_id = id;
    return this;
  }

  /**
   * 设置外边距
   */
  margin(margin: string): ChartBuilder {
    this.chart.margin = margin;
    return this;
  }

  /**
   * 构建 Chart 对象
   */
  build(): Chart {
    return this.chart;
  }
}
